wcfsetup/install/files/acp/update_com.woltlab.wcf_5.4_session_1_cookies.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Sets the new session cookies.
+ * 
+ * @author	Tim Duesterhus
+ * @copyright	2001-2020 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package	WoltLabSuite\Core
+ */
+
+use wcf\system\application\ApplicationHandler;
+use wcf\system\form\container\GroupFormElementContainer;
+use wcf\system\form\element\LabelFormElement;
+use wcf\system\form\FormDocument;
+use wcf\system\request\RouteHandler;
+use wcf\system\WCF;
+use wcf\util\CryptoUtil;
+use wcf\util\HeaderUtil;
+
+// 1) Check whether the cookies are already in place.
+$sessionIdOkay = false;
+if (!empty($_COOKIE[COOKIE_PREFIX."acp_session"])) {
+	$cookieValue = CryptoUtil::getValueFromSignedString($_COOKIE[COOKIE_PREFIX."acp_session"]);
+	if ($cookieValue) {
+		$sessionID = \bin2hex($cookieValue);
+		if ($sessionID === WCF::getSession()->sessionID) {
+			$sessionIdOkay = true;
+		}
+	}
+}
+
+$xsrfTokenOkay = false;
+if (!empty($_COOKIE['XSRF-TOKEN'])) {
+	if (CryptoUtil::validateSignedString($_COOKIE['XSRF-TOKEN'])) {
+		$xsrfTokenOkay = true;
+	}
+}
+
+if ($sessionIdOkay && $xsrfTokenOkay) {
```

Comment:
Please rename these variables to `$hasValidSessionCookie` and `$hasValidXsrfToken` to be more in line with our naming patterns.